backend/dataall/modules/dataset_sharing/api/queries.py

```diff
@@ -61,3 +62,11 @@
     type=gql.ArrayType(gql.Ref('SharedDatasetTableItem')),
     resolver=list_shared_tables_by_env_dataset,
 )
+
+getDatasetSharedAssumeRoleUrl = gql.QueryField(
```

Comment:
This is a generic comment about how the conditional module loading works and the feature flag works.

I see that this Query relies on the `get_dataset_shared_assume_role_url` resolver but this is enabled conditionally. Does it mean that the user can call this query but will receive `Disabled by config ...`? 
Wouldn't it better if we don't define the query at all?

Reply:
I see, so having the same `@is_feature_enabled('modules.datasets.features.aws_actions')` logic on the Query creation instead of in the resolver call right? Yes, it does make sense and it might be possible with AppSync I guess as we define API calls as infra as code independently right?

Reply:
Yes that's what I mean, I think it's possible here, instead of putting the decorator in the resolver you can put an `if` in the definition of the QueryField.